### DISCUSSION TITLE
4.6.4

### DISCUSSION THREAD

**Discussion by ywmoyue** [ywmoyue/biliuwp-lite#170] [Announcements]
#### CHANGED 
- 添加评论请求错误备用方案
- 支持视频播放和视频下载选择音质
#### FIXED
- 修复导出视频失败

<hr /><em>This discussion was created from the release <a href='https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.4'>4.6.4</a>.</em>

**Comment by AvalonUltra** [comment]
终于可以选择音质了，好评，堪称史诗级更新！但是我貌似在支持杜比全景声的视频里没看见杜比全景声的选项？比如这个视频https://www.bilibili.com/video/BV1eV411W7tt
电脑本身没有杜比音效认证，但是买了MS商店里的Dolby Acess, Netflix是可以点亮Dolby Atmos标的。不知道是我的设备问题还是目前没有支持？

**Reply by ywmoyue** [reply]
没支持，没找到哪个视频的查询playurl的接口有返回杜比全景声的数据，可能需要使用另外的接口
![image](https://github.com/ywmoyue/biliuwp-lite/assets/28388300/5d770f21-87d0-4ff7-bad8-7fbb08178d93)

**Reply by AvalonUltra** [reply]
我这边不是null
![image](https://github.com/ywmoyue/biliuwp-lite/assets/66233248/232fe2a2-a118-431d-ae7f-3928ceb18a6f)
我是在Edge浏览器里改了UA为MacOS的Safari15.4，这样就可以出现Dolby Atmos的选项了。不管有没有用我先把这个复制上来，你可以看着用一下
"dolby":{"type":2,"audio":[{"id":30250,"baseUrl":"https://cn-sdjn-fx-01-01.bilivideo.com/upgcxcode/93/66/375186693/375186693-1-30250.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1687820089&gen=playurlv2&os=bcache&oi=985835986&trid=0000b878a72e79b54758963b57797629d5e4u&mid=435816551&platform=pc&upsig=7b977379fb7ee2e1cdb617f3f45d0712&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&cdnid=6801&bvc=vod&nettype=0&orderid=0,3&buvid=26365FDA-F31E-BE09-3A69-68C0A04B9F4F61856infoc&build=0&agrr=1&bw=80428&np=151339420&logo=80000000","base_url":"https://cn-sdjn-fx-01-01.bilivideo.com/upgcxcode/93/66/375186693/375186693-1-30250.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1687820089&gen=playurlv2&os=bcache&oi=985835986&trid=0000b878a72e79b54758963b57797629d5e4u&mid=435816551&platform=pc&upsig=7b977379fb7ee2e1cdb617f3f45d0712&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&cdnid=6801&bvc=vod&nettype=0&orderid=0,3&buvid=26365FDA-F31E-BE09-3A69-68C0A04B9F4F61856infoc&build=0&agrr=1&bw=80428&np=151339420&logo=80000000","backupUrl":["https://upos-sz-mirrorali.bilivideo.com/upgcxcode/93/66/375186693/375186693-1-30250.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1687820089&gen=playurlv2&os=alibv&oi=985835986&trid=b878a72e79b54758963b57797629d5e4u&mid=435816551&platform=pc&upsig=812e5b8d34f5af8b7212ba15efbd81c9&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&bvc=vod&nettype=0&orderid=1,3&buvid=26365FDA-F31E-BE09-3A69-68C0A04B9F4F61856infoc&build=0&agrr=1&bw=80428&np=151339420&logo=40000000","https://upos-sz-mirrorali.bilivideo.com/upgcxcode/93/66/375186693/375186693-1-30250.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1687820089&gen=playurlv2&os=alibv&oi=985835986&trid=b878a72e79b54758963b57797629d5e4u&mid=435816551&platform=pc&upsig=812e5b8d34f5af8b7212ba15efbd81c9&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&bvc=vod&nettype=0&orderid=2,3&buvid=26365FDA-F31E-BE09-3A69-68C0A04B9F4F61856infoc&build=0&agrr=1&bw=80428&np=151339420&logo=40000000"],"backup_url":["https://upos-sz-mirrorali.bilivideo.com/upgcxcode/93/66/375186693/375186693-1-30250.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1687820089&gen=playurlv2&os=alibv&oi=985835986&trid=b878a72e79b54758963b57797629d5e4u&mid=435816551&platform=pc&upsig=812e5b8d34f5af8b7212ba15efbd81c9&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&bvc=vod&nettype=0&orderid=1,3&buvid=26365FDA-F31E-BE09-3A69-68C0A04B9F4F61856infoc&build=0&agrr=1&bw=80428&np=151339420&logo=40000000","https://upos-sz-mirrorali.bilivideo.com/upgcxcode/93/66/375186693/375186693-1-30250.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1687820089&gen=playurlv2&os=alibv&oi=985835986&trid=b878a72e79b54758963b57797629d5e4u&mid=435816551&platform=pc&upsig=812e5b8d34f5af8b7212ba15efbd81c9&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&bvc=vod&nettype=0&orderid=2,3&buvid=26365FDA-F31E-BE09-3A69-68C0A04B9F4F61856infoc&build=0&agrr=1&bw=80428&np=151339420&logo=40000000"],"bandwidth":640202,"mimeType":"audio/mp4","mime_type":"audio/mp4","codecs":"ec-3","width":0,"height":0,"frameRate":"","frame_rate":"","sar":"","startWithSap":0,"start_with_sap":0,"SegmentBase":{"Initialization":"0-821","indexRange":"822-1309"},"segment_base":{"initialization":"0-821","index_range":"822-1309"},"codecid":0}]}

**Reply by TPAXcc** [reply]
> 没支持，没找到哪个视频的查询playurl的接口有返回杜比全景声的数据，可能需要使用另外的接口
> ![image](https://github.com/ywmoyue/biliuwp-lite/assets/28388300/5d770f21-87d0-4ff7-bad8-7fbb08178d93)
> 
https://github.com/SocialSisterYi/bilibili-API-collect/blob/master/docs/video/videostream_url.md#视频伴音音质代码

**Reply by ywmoyue** [reply]
4.6.7已提供杜比全景声选择，麻烦验证一下

**Reply by AvalonUltra** [reply]
> 4.6.7已提供杜比全景声选择，麻烦验证一下

验证过了，没有问题，可以开启杜比全景声并点亮显示器的杜比全景声标志

**Comment by Playfulnurse** [comment]
如何启用评论备用方案呢，本来没有评论请求失败的，更新后每个视频打开后都是评论请求失败

**Comment by Shaddoll126** [comment]
今天晚上又评论请求失败了   备用方案似乎失效了

**Comment by ywmoyue** [comment]
非常抱歉写了bug，备用方案没生效 @Shaddoll126  @Playfulnurse ，我修复一下

**Reply by Playfulnurse** [reply]
非常感谢回复，麻烦了

**Reply by ywmoyue** [reply]
https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.4.2 
已修复，先不发正式版了 @Shaddoll126 @Playfulnurse

**Comment by jnbadnsodkns** [comment]
还存在加载评论失败：请求错误的问题

**Reply by ywmoyue** [reply]
请将日志文件通过邮件发送给我

**Reply by ywmoyue** [reply]
日志文件我已看过，看着不像是4.6.4.2版本以上的日志，请安装最新的临时版本https://github.com/ywmoyue/biliuwp-lite/releases
